source/agora/node/BlockStorage.d

```diff
@@ -932,8 +924,9 @@ public class MemBlockStorage : IBlockStorage
 
     public void saveBlock (const ref Block block) @safe
     {
-        if (this.blocks.length != block.header.height)
-            throw new Exception("BlockStorage: Expected blocks in serial order");
+        ensure(this.blocks.length == block.header.height,
+                "BlockStorage: Expected blocks in serial order (known: {}, got: {})",
```

Comment:
```suggestion
                "BlockStorage: Expected block at height {} not at height {}",
```

Reply:
The reason "serial order" is here is to inform the user not only of what is expected, but most importantly, why it is expected. Here it's because we require to `saveBlock` height `N` before we `saveBlock` height `N + 1`.
I agree that the terms are not ideal though, so maybe there's a better way to carry this to the user ?

Reply:
```suggestion
                "BlockStorage: Expected blocks in serial order (next height is {} not {}",
```